spec/rubocop/ast/node_spec.rb

```diff
@@ -347,4 +347,30 @@ def used?
       end
     end
   end
+
+  describe '#argument_type?' do
+    context 'block arguments' do
+      let(:src) { 'bar { |a, b = 42, *c, d: 42, **e| nil }' }
+
+      it 'returns true for all argument types' do
+        node.children[1].children.each do |arg|
```

Comment:
You can use `arguments` instead of `children[1]`, makes it even more clear :-)

Reply:
You may also be able to write this as:

```ruby
expect(node.arguments.children.all?(&:argument_type?)).to eq(true)
```

...but more of a nit

Reply:
Hopefully we'll soon be writing `.to all(be_argument_type)` [see this](https://github.com/rubocop-hq/rubocop-rspec/issues/919)